### PR TITLE
klaw-sync: upgrade types to match v5.0.0 API

### PR DESCRIPTION
**Pull request by AviVahl**
v4 removed `noRecurseOnFailedFilter` and added `fs` options
v5 added `depthLimit`

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/manidlou/node-klaw-sync
https://github.com/manidlou/node-klaw-sync/blob/master/CHANGELOG.md
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by AviVahl** [comment]
master is currently broken. PR got same CI break (relating to #28759).
FYI @RyanCavanaugh

File: types/klaw-sync/klaw-sync-tests.ts [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
+import * as fs from 'fs'
 
 const outputMessage = (result: klawSync.Item) => {
     console.log(`file: ${result.path} has size '${result.stats.size}'`)
 }
 
 klawSync('/some/dir').forEach(outputMessage)
 
-const defaultOptions = {}
+const defaultOptions: klawSync.Options = {}
```
Comment (shiftkey) [inline types/klaw-sync/klaw-sync-tests.ts]: You might be able to avoid declaring `klawSync.Options` here and below. Did you encounter a specific error?

Comment (AviVahl) [reply types/klaw-sync/klaw-sync-tests.ts]: I actually added the explicit type to be more consistent with the change for the `const options` below.

I wanted to make sure the test isn't passing non-`Options` keys.

Before I added the explicit type, nothing ensured `noRecurseOnFailedFilter` is not a valid `Options` key, as the type of `options` was inferred and compatible with the call to `klawSync()`.

This isn't a must, but gives a bit of added value IMO. Would you like me to revert this back?

Comment (shiftkey) [reply types/klaw-sync/klaw-sync-tests.ts]: Aha, I missed that detail in the test below. All good!

**Review by shiftkey — APPROVED** [review]

**Pull request merged**